<a href="https://colab.research.google.com/github/artbrgn/DS-Unit-1-Sprint-3-Statistical-Tests-and-Experiments/blob/master/module1-statistics-probability-and-inference/Arturo_Obregon_LS_DS_131_Statistics_Probability_Assignment.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

<img align="left" src="https://lever-client-logos.s3.amazonaws.com/864372b1-534c-480e-acd5-9711f850815c-1524247202159.png" width=200>
<br></br>
<br></br>

## *Data Science Unit 1 Sprint 3 Assignment 1*

# Apply the t-test to real data

Your assignment is to determine which issues have "statistically significant" differences between political parties in this [1980s congressional voting data](https://archive.ics.uci.edu/ml/datasets/Congressional+Voting+Records). The data consists of 435 instances (one for each congressperson), a class (democrat or republican), and 16 binary attributes (yes or no for voting for or against certain issues). Be aware - there are missing values!

Your goals:

1. Load and clean the data (or determine the best method to drop observations when running tests)
2. Using hypothesis testing, find an issue that democrats support more than republicans with p < 0.01
3. Using hypothesis testing, find an issue that republicans support more than democrats with p < 0.01
4. Using hypothesis testing, find an issue where the difference between republicans and democrats has p > 0.1 (i.e. there may not be much of a difference)

Note that this data will involve *2 sample* t-tests, because you're comparing averages across two groups (republicans and democrats) rather than a single group against a null hypothesis.

Stretch goals:

1. Refactor your code into functions so it's easy to rerun with arbitrary variables
2. Apply hypothesis testing to your personal project data (for the purposes of this notebook you can type a summary of the hypothesis you formed and tested)

In [1]:
### YOUR CODE STARTS HERE
!wget https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data

--2019-09-16 20:16:34--  https://archive.ics.uci.edu/ml/machine-learning-databases/voting-records/house-votes-84.data
Resolving archive.ics.uci.edu (archive.ics.uci.edu)... 128.195.10.252
Connecting to archive.ics.uci.edu (archive.ics.uci.edu)|128.195.10.252|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 18171 (18K) [application/x-httpd-php]
Saving to: ‘house-votes-84.data’

house-votes-84.data 100%[===================>]  17.75K  --.-KB/s    in 0.03s   

2019-09-16 20:16:34 (586 KB/s) - ‘house-votes-84.data’ saved [18171/18171]



In [29]:
import pandas as pd
import numpy as np
df = pd.read_csv('house-votes-84.data', 
                 header=None,
                 names=['party','handicapped-infants','water-project',
                          'budget','physician-fee-freeze', 'el-salvador-aid',
                          'religious-groups','anti-satellite-ban',
                          'aid-to-contras','mx-missile','immigration',
                          'synfuels', 'education', 'right-to-sue','crime','duty-free',
                          'south-africa'])
print(df.shape)
df.head(1000)

(435, 17)


party handicapped-infants water-project  ... crime duty-free south-africa
0    republican                   n             y  ...     y         n            y
1    republican                   n             y  ...     y         n            ?
2      democrat                   ?             y  ...     y         n            n
3      democrat                   n             y  ...     n         n            y
4      democrat                   y             y  ...     y         y            y
5      democrat                   n             y  ...     y         y            y
6      democrat                   n             y  ...     y         y            y
7    republican                   n             y  ...     y         ?            y
8    republican                   n             y  ...     y         n            y
9      democrat                   y             y  ...     n         ?            ?
10   republican                   n             y  ...     y         n            n
11   republican                   n             y  ...     y         ?            ?
12     democrat                   n             y  ...     n         ?            ?
13     democrat                   y             y  ...     n         y            ?
14   republican                   n             y  ...     ?         n            ?
15   republican                   n             y  ...     ?         n            ?
16     democrat                   y             n  ...     n         n            y
17     democrat                   y             ?  ...     n         y            y
18   republican                   n             y  ...     y         n            n
19     democrat                   y             y  ...     n         y            y
20     democrat                   y             y  ...     n         y            y
21     democrat                   y             y  ...     ?         y            y
22     democrat                   y             ?  ...     n         y            y
23     democrat                   y             y  ...     n         y            y
24     democrat                   y             n  ...     n         y            ?
25     democrat                   y             n  ...     n         y            y
26     democrat                   y             n  ...     n         y            y
27     democrat                   y             y  ...     n         y            y
28   republican                   y             n  ...     y         n            y
29     democrat                   y             y  ...     n         y            y
..          ...                 ...           ...  ...   ...       ...          ...
405  republican                   n             n  ...     y         n            y
406    democrat                   y             n  ...     y         n            y
407    democrat                   n             n  ...     y         n            n
408    democrat                   y             n  ...     y         y            y
409  republican                   n             n  ...     y         n            n
410  republican                   n             n  ...     y         n            y
411    democrat                   y             n  ...     n         n            y
412  republican                   n             n  ...     y         n            y
413  republican                   y             y  ...     y         n            y
414    democrat                   y             y  ...     n         n            y
415    democrat                   n             y  ...     n         n            y
416  republican                   y             y  ...     y         n            y
417    democrat                   y             y  ...     n         n            y
418    democrat                   y             y  ...     n         n            y
419    democrat                   y             y  ...     n         n            y
420  republican                   y             y  ...     y  

In [8]:
df = df.replace({'?':np.NaN, 'n':0, 'y':1})

df.head()

party  handicapped-infants  ...  duty-free  south-africa
0  republican                  0.0  ...        0.0           1.0
1  republican                  0.0  ...        0.0           NaN
2    democrat                  NaN  ...        0.0           0.0
3    democrat                  0.0  ...        0.0           1.0
4    democrat                  1.0  ...        1.0           1.0

[5 rows x 17 columns]

In [0]:
dem = df[df['party'] == 'democrat']
rep = df[df['party'] == 'republican']

In [10]:
dem.groupby(['party']).sum()

handicapped-infants  water-project  ...  duty-free  south-africa
party                                         ...                         
democrat                156.0          120.0  ...      160.0         173.0

[1 rows x 16 columns]

In [11]:
rep.groupby(['party']).sum()

handicapped-infants  water-project  ...  duty-free  south-africa
party                                           ...                         
republican                 31.0           75.0  ...       14.0          96.0

[1 rows x 16 columns]

In [0]:
from scipy.stats import ttest_1samp

In [14]:
rep['immigration'].mean()

0.5575757575757576

In [15]:
rep['immigration'].isnull().sum()

3

In [17]:
len(rep['immigration']) - rep['immigration'].isnull().sum()

165

1) Null Hypothesis:

In 1-sample t-tests YOU GET TO CHOOSE YOUR NULL HYPOTHESIS

H0 : 0.0 - There is ZERO republican support for this bill

2) Alternative Hypothesis

Ha :  x¯≠0  - There is non-zero support for the budget bill among repulbicans.

3) Confidence Level: 95% or .95

In [18]:
ttest_1samp(rep['immigration'], 0, nan_policy = 'omit')

Ttest_1sampResult(statistic=14.376541013291384, pvalue=7.541248569126767e-31)

4) t-statistic: 14.3765

5) p-value of .00000000000000000000000000007541
______________________________________________________________________

Concluision: Due to a p-value of 0 I reject the null hypothesis that republican support is zero and conclude that republican support is non-zero.



1) Null Hypothesis:

In 1-sample t-tests YOU GET TO CHOOSE YOUR NULL HYPOTHESIS

H0 : 0.0 - There is ZERO democratic support for this bill

2) Alternative Hypothesis

Ha : x¯≠0 - There is non-zero support for the budget bill among democratics.

3) Confidence Level: 95% or .95

In [19]:
ttest_1samp(dem['synfuels'], 0, nan_policy = 'omit')

Ttest_1sampResult(statistic=16.125992032976423, pvalue=9.159793118867048e-41)

4) t-statistic: 16.1259

5) p-value of .000000000000000000000000000000000000009159
______________________________________________________________________

Concluision: Due to a p-value of 0 I reject the null hypothesis that democratic support is zero and conclude that democratic support is non-zero.

---
---
2-SAMPLE TEST


In [0]:
from scipy.stats import ttest_ind
import numpy as np

In [25]:
ttest_ind(rep['education'], dem['education'], nan_policy = 'omit')

Ttest_indResult(statistic=20.500685724563073, pvalue=1.8834203990450192e-64)

In [27]:
rep['education'].mean()

0.8709677419354839

In [28]:
dem['education'].mean()

0.14457831325301204